smartsim/entity/dbobject.py

```diff
@@ -53,6 +54,7 @@ def __init__(
             self.file = self._check_filepath(file_path)
         self.device = self._check_device(device)
         self.devices_per_node = devices_per_node
+        self._check_arguments()
```

Comment:
Consider using `device: Literal["CPU", "GPU"]` type hint to tighten the arguments for device (instead of `str`) in `_check_device`/`__init__`